utils/rjson.hh

```diff
@@ -81,6 +81,8 @@ namespace internal {
 class throwing_allocator : public rapidjson::CrtAllocator {
     using base = rapidjson::CrtAllocator;
 public:
+    // For bypassing the address sanitizer failure in debug mode
+    static constexpr size_t allocation_failure_threshold_in_debug_mode = 0xffffafffffff;
```

Comment:
This number is weird and I wonder if it will not change between compilers and versions, but we can do this for now.
I was thinking about something less arbitrary (?) - Seastar know the maximum amount of memory we have (the "-m" option), e.g., 1GB. This memory limit isn't actually used in the sanitizer build, but we still have this as an estimate, and can still fail any single allocation of more than that 1GB limit. It doesn't really need to be some absurdly huge number.
But that being said, your approach is ok too. Until the compiler changes, I guess.

Reply:
Good idea. We do have `memory::stats().total_memory()`, so it's better to get rid of this artificial constant and use that. I'll post v2 soon